src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -119,32 +121,51 @@ void ObjectBufferPool::WriteChunk(const ObjectID &object_id,
                                   uint64_t metadata_size,
                                   const uint64_t chunk_index,
                                   const std::string &data) {
-  absl::MutexLock lock(&pool_mutex_);
-  auto it = create_buffer_state_.find(object_id);
-  if (it == create_buffer_state_.end() || chunk_index >= it->second.chunk_state.size() ||
-      it->second.chunk_state.at(chunk_index) != CreateChunkState::REFERENCED) {
-    RAY_LOG(DEBUG) << "Object " << object_id << " aborted before chunk " << chunk_index
-                   << " could be sealed";
-    return;
-  }
-  if (it->second.data_size != data_size || it->second.metadata_size != metadata_size) {
-    RAY_LOG(DEBUG) << "Object " << object_id << " size mismatch, rejecting chunk";
-    return;
+  std::optional<ChunkInfo> chunk_info;
```

Comment:
[ChunkInfo](https://github.com/ray-project/ray/blob/master/src/ray/object_manager/object_buffer_pool.h#L41) is an offest based structure, where it contains a `uint8_t* data` and a `std::shared_ptr<Buffer>`. `data` simply indexes into the `Buffer` at an offset. 

Which means as long as a `ChunkInfo` copy is held by the function, the underlying buffer is not released.